### ISSUE TITLE
run_on_start missing on salt.states.schedule

### ISSUE THREAD

**Issue by cmclaughlin** [saltstack/salt#35420]
### Description of Issue/Question

run_on_start seems to be unsupported on on salt.states.schedule

It seems only to be supported when defining schedules via pillar data

(On a side note, defining schedules from pillar data requires minions to restart (?)... so I prefer to define schedules in states, where schedule changes take immediate effect)

### Setup

Define a state such as this:

	my_highstate:
	  schedule.present:
	    - function: my_function
	    - run_on_start: False

### Steps to Reproduce Issue
Apply the state and list the schedule... run_on_start won't be disabled on the job. 

### Versions Report
	Salt Version:
	           Salt: 2015.8.10

	Dependency Versions:
	         Jinja2: 2.7.2
	       M2Crypto: Not Installed
	           Mako: 0.9.1
	         PyYAML: 3.10
	          PyZMQ: 14.0.1
	         Python: 2.7.6 (default, Mar 22 2014, 22:59:56)
	           RAET: Not Installed
	        Tornado: 4.2.1
	            ZMQ: 4.0.4
	           cffi: Not Installed
	       cherrypy: 3.2.2
	       dateutil: 2.4.2
	          gitdb: 0.5.4
	      gitpython: 0.3.2 RC1
	          ioflo: Not Installed
	        libgit2: Not Installed
	        libnacl: Not Installed
	   msgpack-pure: Not Installed
	 msgpack-python: 0.3.0
	   mysql-python: 1.2.3
	      pycparser: Not Installed
	       pycrypto: 2.6.1
	         pygit2: Not Installed
	   python-gnupg: 0.3.6
	          smmap: 0.8.2
	        timelib: Not Installed

	System Versions:
	           dist: Ubuntu 14.04 trusty
	        machine: x86_64
	        release: 3.13.0-48-generic
	         system: Ubuntu 14.04 trusty

**Comment by Ch3LL** [comment]
@cmclaughlin I am seeing the same behavior with `run_on_start: False` not being in the output of `schedule.list` when adding it to the state module, so I will add this as a bug. 

The other issue though I am seeing is if its set in pillar I do not see it in schedule.list either using this exampel:

```
schedule:
  job1:
    function: state.sls
    seconds: 3600
    run_on_start: False
    args:
      - httpd
    kwargs:
      test: True
```

So my question is what pillar example are you using where you are seeing it in the output of listing jobs?

**Comment by cmclaughlin** [comment]
@Ch3LL thanks... here's the schedule I had defined in pillar:

	schedule:
	  highstate:
	    function: state.highstate
	    minutes: 240
	    splay: 600  # spread out runs 0 to 600 seconds (10m)
	    kwargs:
	      test: True
	      saltenv: base
	      pillarenv: base
	    returner: highstate  # custom returner that emails pending changes

While using this config, I found I needed to restart the minion before it would show up in schedule.list.  I'm not sure if that's expected or not... but I think we both agree it's not optimal.  At first I thought there was something wrong with my Pillar code... then I found these docs and figured restarting the minion would be worth a shot...

From https://docs.saltstack.com/en/latest/topics/jobs/#scheduling-jobs

_"Schedules that are implemented via pillar data, only need to refresh the minion's pillar data, for example by using saltutil.refresh_pillar. Schedules implemented in the master or minion config have to restart the application in order for the schedule to be implemented."_

After restarting the minion, my schedule defined via Pillar did show up.  It doesn't sound like that should be necessary, so maybe we should open another bug report.

Also I couldn't reliably edit/change the schedule via Pillar.... I originally defined the schedule in Pillar with seconds, but tried to remove seconds and replace it with minutes.  The schedule changed every time I ran highstate, even though there shouldn't be any changes.  I also had problems deleting the schedule when it was defined via Pillar.  All that led to me replacing it with the state definition.

I've been much happier with the way schedules work when defining it as a state... I can edit/change it reliably.  Supporting run_on_start via the state would be nice, hence this issue... but overall I've just found defining schedules as states to work as expected.

**Comment by Ch3LL** [comment]
@cmclaughlin yes if you can open another issue if your schedule is still not showing up after refreshing the pillar data and same with the other issue related to the information not updating correctly with pillar data and we can track them there. Thanks for the additional information. 

As a note I was not able to replicate the schedule not showing up in `schedule.list` after running `salt '*' saltutil.refresh_pillar` so please include a small test case in the issue you open and we can track it there and I'll try to replicate it thanks.

**Comment by sastorsl** [comment]
I'm seeing the same thing re "run_at_start"
There is no `run_at_start` in `schedule.present` - https://docs.saltstack.com/en/latest/ref/states/all/salt.states.schedule.html#salt.states.schedule.present
Nor does the module `schedule.add` seem to heed `run_at_start=False`
https://docs.saltstack.com/en/2015.8/ref/modules/all/salt.modules.schedule.html#salt.modules.schedule.add

**Comment by Ch3LL** [comment]
@sastorsl i'm confused, I do not see an option `run_at_start` anywhere in salt's code. Can you clarify?

**Comment by cmclaughlin** [comment]
@sastorsl you must mean run_on_start, not run_at_start...?  That said, I think you're just agreeing with the original issue.

**Comment by sastorsl** [comment]
Typo on my part. Sorry about the confusion. And yes,  I was just confirming that this was an issue for others/me too.

**Comment by Ch3LL** [comment]
Awesome. Thanks for clarifying :) Would anyone be willing to give the fix in #35586 and see if that resolves the issue for them? And then we can get this issue closed up. Thanks

**Comment by cmclaughlin** [comment]
I just upgraded to 2016.3.3 and can verify this.  Thanks!
